src/thumbnailer.cc

```diff
@@ -39,16 +39,80 @@ Thumbnailer::~Thumbnailer() { WebPAnimEncoderDelete(enc_); }
 Thumbnailer::Status Thumbnailer::AddFrame(const WebPPicture& pic,
                                           int timestamp_ms) {
   // Verify dimension of frames.
-  if (!frames_.empty() && (pic.width != (frames_[0].pic).width ||
-                           pic.height != (frames_[0].pic).height)) {
+  if (!frames_.empty() && (pic.width != frames_[0].pic.width ||
+                           pic.height != frames_[0].pic.height)) {
     return kImageFormatError;
   }
   WebPConfig new_config;
-  WebPConfigInit(&new_config);
+  assert(WebPConfigInit(&new_config));
+  new_config.show_compressed = 1;
   frames_.push_back({pic, timestamp_ms, new_config});
   return kOk;
 }
 
+int Thumbnailer::GetSize(WebPPicture* const pic, const WebPConfig& config) {
+  int failure = -1;
```

Comment:
This constant (`failure`) will be returned by functions that should return nonnegative integer instead. Should we declare an `enum kUnsignedError = -1`?

Reply:
You may although `enum`s are typically used for several meanings (`Status::kOk`, `Status::kMemoryError` etc.) and not for actual numbers (-1).
I would suggest `static const int kFailure = -1;` or using the already existing `Thumbnailer::Status` everywhere instead (requires adding output arguments to `GetPSNR()`, `GetSize()` because the returned value will now only be the status).